planner/core/rule_predicate_push_down.go

```diff
@@ -391,7 +391,7 @@ func (p *LogicalProjection) PredicatePushDown(predicates []expression.Expression
 	}
 	for _, cond := range predicates {
 		newFilter := expression.ColumnSubstitute(cond, p.Schema(), p.Exprs)
-		if !expression.HasGetSetVarFunc(newFilter) {
+		if !expression.HasGetSetVarFunc(newFilter) && expression.ExprFromSchema(newFilter, p.Children()[0].Schema()) {
```

Comment:
Any clearer comment why the `t2.b > 'a'` doesn't match the newly added `ExprFromSchema`?

Reply:
Because the Projection creates a new column named `Column#7` to present `'a' collate utf8mb4_unicode_ci`, and `Column#7` cannot be substituted in `ColumnSubstitute`([L268](https://github.com/pingcap/tidb/blob/master/expression/util.go#L268)).
So if we push `Column#7` down, the underlying operators cannot recognize it when `ResolveIndex`.
This check seems horrible that may cause many regressions, but actually it only affects columns with specified collation.
Do you have any suggestions about this issue? @winoros @xuyifangreeneyes

Reply:
Oh, after i go through the codes.
I found that the `ColumnSubsititute` might fail when correlation is enabled.
We can use `ColumnSubstituteImpl` to get the bool value which tells us whether the expr's substitution success.

Reply:
I'll try to remove this logic(It might substitute fail) in the future.